packages/client/src/lib.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * A client library for the https://filecoin.storage/ service. It provides a convenient
+ * interface for working with the [Raw HTTP API](https://filecoin.storage/#api-docs)
+ * from a web browser or [Node.js](https://nodejs.org/) and comes bundled with
+ * TS for out-of-the box type inference and better IntelliSense.
+ *
+ * @example
+ * ```js
+ * import { FilecoinStorage, Blob } from "filecoin.storage"
+ * const client = new FilecoinStorage({ token: API_TOKEN })
+ *
+ * const cid = await client.storeBlob(new Blob(['hello world']))
+ * ```
+ * @module
+ */
+import * as API from './lib/interface.js'
+import { fetch, Blob } from './platform.js'
+
+/**
+ * @implements API.Service
+ */
+class FilecoinStorage {
+  /**
+   * Constructs a client bound to the given `options.token` and
+   * `options.endpoint`.
+   *
+   * @example
+   * ```js
+   * import { FilecoinStorage, Blob } from "filecoin.storage"
+   * const client = new FilecoinStorage({ token: API_TOKEN })
+   * const { car, rootCid } = await client.pack(new Blob(['hello world']))
+   * const cid = await client.store(car)
+   * console.assert(cid === rootCid, 'The service should store the files with the `rootCid` I created')
+   * ```
+   * Optionally you could pass an alternative API endpoint (e.g. for testing)
+   * @example
+   * ```js
+   * import { FilecoinStorage } from "filecoin.storage"
+   * const client = new FilecoinStorage({
+   *   token: API_TOKEN
+   *   endpoint: new URL('http://localhost:8080/')
+   * })
+   * ```
+   *
+   * @param {{token: string, endpoint?:URL}} options
+   */
+  constructor({ token, endpoint = new URL('https://api.filecoin.storage') }) {
+    /**
+     * Authorization token.
+     *
+     * @readonly
+     */
+    this.token = token
+    /**
+     * Service API endpoint `URL`.
+     * @readonly
+     */
+    this.endpoint = endpoint
+  }
+
+  /**
+   * @hidden
+   * @param {string} token
+   */
+  static headers(token) {
+    if (!token) throw new Error('missing token')
+    return {
+      Authorization: `Bearer ${token}`,
+      'X-Client': 'filecoin.storage',
```

Comment:
Add a custom header so we can figure out how many folks actually use this thing

Reply:
Are we interested in version or language?

Reply:
Let's start with this and I'll ask around to see what folks think. I'm inclined to track less.